src/components/marquee-wizard/marquee-wizard.component.spec.ts

```diff
@@ -273,3 +282,137 @@ describe('Marquee wizard with delayed step creation', () => {
+            content: 'Content of step 2.',
+        },
+        {
+            header: '3. Third Step',
+            content: 'Content of step 3.',
+        },
+        {
+            header: '4. Fourth Step',
+            content: 'Content of step 4.',
+        }
+    ];
+    stepChanging(_: StepChangingEvent) {}
+    stepChange(_: number) {}
+    onNext(_: number) {}
+    visitedChanged(_: boolean) { }
+
+    @ViewChildren(MarqueeWizardStepComponent)
+    stepsList: QueryList<MarqueeWizardStepComponent>;
+}
+
+describe('Marquee wizard with visitedChange event', () => {
+    let component: MarqueeWizardVisitedChangeTestComponent;
+    let fixture: ComponentFixture<MarqueeWizardVisitedChangeTestComponent>;
+    let nativeElement: HTMLElement;
+    let visitedChanged: jasmine.Spy;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [MarqueeWizardModule, CommonModule],
+            declarations: [MarqueeWizardVisitedChangeTestComponent]
+        }).compileComponents();
+    }));
+
+    beforeEach(async () => {
+        fixture = TestBed.createComponent(MarqueeWizardVisitedChangeTestComponent);
+        component = fixture.componentInstance;
+        nativeElement = fixture.nativeElement;
+        visitedChanged = spyOn(component, 'visitedChanged');
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+        fixture.detectChanges();
+        await fixture.whenStable();
+    });
+
+    it('should trigger a visitedChange event when valid modified on the current step, when other steps ahead are visited', async () => {
+        // set step 1 to valid and move forward
+        await clickButton(MarqueeWizardSelectors.ToggleValidity);
+        await clickButton(MarqueeWizardSelectors.NextButton);
+
+        // set step 2 to valid and move forward
+        await clickButton(MarqueeWizardSelectors.ToggleValidity);
+        await clickButton(MarqueeWizardSelectors.NextButton);
+
+        // set step 3 to valid and move forward
+        await clickButton(MarqueeWizardSelectors.ToggleValidity);
+        await clickButton(MarqueeWizardSelectors.NextButton);
+
+        // jump back to the first step
+        await clickButton(MarqueeWizardSelectors.SecondStep);
+
+        visitedChanged.calls.reset();
+
+        // valid now false and should trigger visitedChange
+        await clickButton(MarqueeWizardSelectors.ToggleValidity);
+
+        await clickButton(MarqueeWizardSelectors.NextButton);
+
+        expect(visitedChanged).toHaveBeenCalledWith(false);
+        expect(visitedChanged).toHaveBeenCalledTimes(1);
+
+        const stepsList = component.stepsList.toArray();
+        // step 1 should be valid and visited
+        expect(stepsList[0]._valid).toBeTruthy();
+        expect(stepsList[0].visited).toBeTruthy();
+        expect(stepsList[0].completed).toBeTruthy();
+
+        // step 2 should be invalid and not visited
+        expect(stepsList[1]._valid).toBeFalsy();
```

Comment:
Could we perhaps add a checks to see if the `visitedChange` has emitted on these steps correctly?